pkg/network/usm/shared_libraries.go

```diff
@@ -145,23 +175,29 @@ type soRegistration struct {
 }
 
 // unregister return true if there are no more reference to this registration
-func (r *soRegistration) unregisterPath(pathID pathIdentifier) bool {
+func (r *soRegistration) unregisterPath(pathID pathIdentifier, soreg *soRegistry) bool {
 	currentUniqueProcessesCount := r.uniqueProcessesCount.Dec()
 	if currentUniqueProcessesCount > 0 {
 		return false
 	}
 	if currentUniqueProcessesCount < 0 {
 		log.Errorf("unregistered %+v too much (current counter %v)", pathID, currentUniqueProcessesCount)
+		soreg.libUnregisterErrors.Add(1)
 		return true
 	}
+
 	// currentUniqueProcessesCount is 0, thus we should unregister.
 	if r.unregisterCB != nil {
 		if err := r.unregisterCB(pathID); err != nil {
 			// Even if we fail here, we have to return true, as best effort methodology.
 			// We cannot handle the failure, and thus we should continue.
 			log.Warnf("error while unregistering %s : %s", pathID.String(), err)
+			soreg.libUnregisterErrors.Add(1)
```

Comment:
everytime we have an error, we also have libUnregistered, I don't think that's correct

Reply:
we still unregister in all cases (as the comment say)
and would make the difference compare to old line 191

Reply:
why not creating a separate metric called 'FailedCB` ?